README.md

```diff
@@ -1,11 +1,212 @@
+| A | A | A | B | B | B | C | C | C |
+
+This situation is not ideal because the same backend must handle multiple
+consecutive requests while the other backends are idle. The ideal for
+cueball is to have an even distribution of work among the backends not just
+with respect to connection count, but also with respect to to request
+distribution over time. Now admittedly the above example is an extreme
+case and the pattern could quickly resolve itself based on the workload, but
+there is no guarantee that would be the case. The goal of the periodic
+decoherence shuffle in cueball is to disrupt these sorts of patterns that
+might arise and persist for an extended period.
+
+There are two configuration options related to decoherence:
+`decoherence_interval` and `decoherence_delay`. The `decoherence_interval`
+represents the length of the period of the decoherence shuffle in seconds. If no
+value is specified for this in the `ConnectionPoolOptions` struct the default
+value is 300 seconds. The `decoherence_delay` is the amount of time in
+milliseconds before the first decoherence shuffle is done once the connection
+pol is created. The default value is 100 milliseconds.
+
+### Example
+
+Use of cueball for connection management requires both an implementation of
+the `Resolver` trait and an implementation
+of the `Connection` trait. Implementers
+of the `Resolver` trait provide information
+to the connection pool about the nodes availble to provide a given
+service. The `Connection` trait defines
+a behavior for establishing and closing a *connection* to a particular
+service.
+
+Here is an example that uses a hypothetical
+`Resolver` and
+`Connection` implementation to create a
+cueball connection pool.
+
+```rust,ignore
+use std::thread;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use std::sync::{Arc, Barrier, Mutex};
+use std::sync::mpsc::Sender;
+use std::{thread, time};
+
+use slog::{Drain, Logger, info, o};
+
+use cueball::backend;
+use cueball::backend::{Backend, BackendAddress, BackendPort};
+use cueball::connection::Connection;
+use cueball::connection_pool::ConnectionPool;
+use cueball::connection_pool::types::ConnectionPoolOptions;
+use cueball::error::Error;
+use cueball::resolver::{BackendAddedMsg, BackendMsg, Resolver};
+
+fn main() {
+    let plain = slog_term::PlainSyncDecorator::new(std::io::stdout());
+    let log = Logger::root(
+        Mutex::new(
+            slog_term::FullFormat::new(plain).build()
+        ).fuse(),
+        o!("build-id" => "0.1.0")
+    );
+
+    let be1 = (IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 55555);
+    let be2 = (IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 55556);
+    let be3 = (IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 55557);
+
+    let resolver = FakeResolver::new(vec![be1, be2, be3]);
+
+    let pool_opts = ConnectionPoolOptions::<FakeResolver> {
+        max_connections: 15,
+        claim_timeout: Some(1000)
+        resolver: resolver,
+        log: log.clone()
```

Comment:
The example should include the decoherence options:

```
            decoherence_interval: None,
            decoherence_delay: None,
```

Reply:
Done